core/services/run_queue.go

```diff
@@ -34,81 +35,180 @@ type RunQueue interface {
 }
 
 type runQueue struct {
-	workersMutex  sync.RWMutex
-	workers       map[string]int
-	workersWg     sync.WaitGroup
-	stopRequested bool
+	runExecutor  RunExecutor
+	workers      map[models.ID]*singleJobSpecWorker // Access to this field is serialized through the select in orchestrateWorkers().
```

Comment:
Does this not have a `len` we can use instead of `numWorkers` below?

Reply:
It does, but then we would have to hook the `WorkerCount()` method into the channel system (probably by having it send on a `requestCount` channel which would be received by `orchestrateWorkers`, and then another channel would be needed to send back the response).